scenarios/test_codegen/test/SerDe_Test.res

```diff
@@ -41,10 +41,25 @@ describe("SerDe Test", () => {
       // arrayOfTimestamps: [],
     }
 
+    //Fails if serialziation does not work
     let set = DbFunctionsEntities.batchSet(~entityMod=module(Entities.EntityWithAllTypes))
-
+    //Fails if parsing does not work
+    let read = DbFunctionsEntities.batchRead(~entityMod=module(Entities.EntityWithAllTypes))
     //set the entity
-    await DbFunctions.sql->set([entity])
+    switch await DbFunctions.sql->set([entity]) {
+    | exception exn =>
+      Js.log(exn)
+      Assert.fail("Failed to set entity in table")
+    | _ => ()
+    }
+
+    switch await DbFunctions.sql->read([entity.id]) {
+    | exception exn =>
+      Js.log(exn)
+      Assert.fail("Failed to read entity from table")
+    | [_entity] => ()
+    | _ => Assert.fail("Should have returned a row on batch read fn")
+    }
```

Comment:
The parsing from the actual entity row was not being tested. Only the entity history which is different since numbers are encoded as strings.

Hasura does return double precision as string but internally direct postgres queries don't.

Reply:
And setting the entities was working fine since the serializer worked perfectly fine on the test.